packages/sdk/src/addresses/index.ts

```diff
@@ -50,3 +52,61 @@ export function getAddressFormat(
   const { type } = getAddressInfo(address);
   return getAddressFormatFromType(type);
 }
+
+function getTaprootAddressFromBip32PublicKey(
+  bip32PublicKey: Buffer,
+  network: Network,
+): Address {
+  const childNodeXOnlyPubkey = bip32PublicKey.subarray(1, 33);
+  const { address } = createPayment(childNodeXOnlyPubkey, "p2tr", network);
+  return {
+    address: address!, // address will never be undefined
+    format: ADDRESS_TYPE_TO_FORMAT.p2tr,
+    publicKey: bip32PublicKey.toString("hex"),
+    xKey: childNodeXOnlyPubkey.toString("hex"),
+  };
+}
+
+export function getAddressesFromPublicKey(
+  publicKey: string | Buffer,
+  network: Network = "mainnet",
+  type: AddressType | "all" = "all",
+): Address[] {
+  const publicKeyBuffer = Buffer.isBuffer(publicKey)
```

Comment:
Can we refactor it that it has fewer returns? There's too much return atm, it's a bit confusing to read if they have the same return

Reply:
okay i'll refactor it again to fix the nesting issue since it can still be simplified